src/auth/components/CallbackPage.tsx

```diff
@@ -1,33 +1,28 @@
-import React, { PureComponent } from "react";
+import React from "react";
 import { withRouter } from "react-router";
-import { CallbackComponent, CallbackComponentProps } from "redux-oidc";
+import { LoginCallbackHandler } from "hds-react";
+import type { OidcClientError, User } from "hds-react";
 import { getRedirectUrlFromSessionStorage } from "@/util/storage";
-import userManager from "@/auth/util/user-manager";
 import { getRouteById, Routes } from "@/root/routes";
+
 type Props = {
   history: Record<string, any>;
 };
 
-const CallbackComponentWithChildren = ({ children, ...rest }: CallbackComponentProps & { children: JSX.Element }): JSX.Element => {
-  return <CallbackComponent {...rest}>
-        {children}
-      </CallbackComponent>;
-}
-
-class CallbackPage extends PureComponent<Props> {
-  successCallback = () => {
-    const {
-      history
-    } = this.props;
+const CallbackPage = (props: Props) => {
+  const onSuccess = (user: User) => {
+    const { history } = props;
     history.push(getRedirectUrlFromSessionStorage() || getRouteById(Routes.LEASES));
```

Comment:
Check if the url startswith `/callback` and don't store that into session storage. Could happen if auth callback from oidc server fails.